### PR TITLE
Remove deprecated code from JRegistry

### PR DESCRIPTION
**Pull request by mbabker**
Per the roadmap, this removes deprecated code in JRegistry.  A couple of additional notes as well:
- Removed commented out methods in JRegistryTest as they had been implemented already elsewhere in the file
- Made a change in JUser::getParameters that I'm not 100% on that calls `$this->_params->loadSetupFile()`, but based on reviewing the old JParameter::loadSetupFile method, I'm fairly certain that the actual intended behavior of this call has been restored (load the file instead of set a boolean true as was happening with JRegistry::loadSetupFile)

### CODE REVIEW

File: tests/suite/joomla/registry/JRegistryTest.php [hunk 1]
```diff
@@ -84,17 +84,6 @@ public function testDef()
 	}
 
 	/**
-	 * @todo Implement testGet().
-	 */
-	/*public function testGet()
```
Comment (chdemko) [inline tests/suite/joomla/registry/JRegistryTest.php]: The get method is not deprecated

File: tests/suite/joomla/registry/JRegistryTest.php [hunk 2]
```diff
@@ -475,18 +403,6 @@ public function testMerge()
...
-		$this->markTestIncomplete(
-		'This test has not been implemented yet.'
-		);
-	}*/
-
-	/**
-	 * Test the JRegistry::set method.
-	 * @deprecated	1.6
-	 */
 	public function testSet()
```
Comment (chdemko) [inline tests/suite/joomla/registry/JRegistryTest.php:406]: The set method is not deprecated

Comment (mbabker) [inline tests/suite/joomla/registry/JRegistryTest.php:406]: This is a commented out test method that is implemented in the method immediately below these deleted lines.
- Michael

Please excuse any errors, this message was sent from my iPhone.

File: tests/suite/joomla/registry/JRegistryTest.php (hunk 1 above)
Comment (mbabker) [inline tests/suite/joomla/registry/JRegistryTest.php]: This too was a duplicate test method that was commented out.  Implemented test is at https://github.com/mbabker/joomla-platform/blob/8204612487005d2a6dfd57fcc65afc2513b0861b/tests/suite/joomla/registry/JRegistryTest.php#L167

**Comment by eddieajau** [comment]
Good stuff!

**Pull request merged**